torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp

```diff
@@ -453,15 +453,8 @@ void ProfilingGraphExecutorImpl::runNoGradOptimizations(
       auto min_size = getFusionGroupInlining() ? 2 : 1;
       bool dyn_shapes = getCurrentBehavior(remaining_bailout_depth) ==
           FusionBehavior::DYNAMIC;
-      FuseTensorExprs(graph, min_size, /*composed_op*/ false, dyn_shapes);
-      GRAPH_DEBUG(
-          "After Fusion, before RemoveTensorTypeSpecializations\n", *graph);
-
-      // Wipe tensor type info from the IR
-      RemoveTensorTypeSpecializations(graph);
-      GRAPH_DEBUG(
-          "After RemoveTensorTypeSpecializations, before customPostPasses\n",
-          *graph);
+      FuseTensorExprs(graph, getFusionGroupInlining() ? 2 : 1);
```

Comment:
We should still be calling `FuseTensorExprs(graph, min_size, /*composed_op*/ false, dyn_shapes);`

You removed the `dyn_shapes` input here